docs/guides/test-isolation.md

```diff
@@ -0,0 +1,104 @@
+# Test Isolation
```

Comment:
was reading through: https://wiki.corp.magento.com/display/QMT/MQE-1355+%5BSPIKE%5D+Identify+Test+Sensitive+Magento+Data and found a couple of sections from it worth mentioning here like about data that could possible affect test outcomes:
1.  data to be restored
2. data to be removed
3. data that shouldn't be modified

snippet from the doc for reference:

> **To Be Restored Test Sensitive Global Effect Entities Data**
There are some Magento entities that affect Magento system globally and should be restored to default value after tests finish. These include:
Index Mode
Store Design Configuration (Content → Design → Configuration) ?

>**To Be Deleted Test Sensitive Global Effect Entities Data**
There are some Magento entities that affect Magento system globally and should be removed after tests finish. These include:
Required product attributes, customer attributes, customer address attributes, returns attributes
Catalog Rules
Cart Price Rules
Related Product Rules

>**Shall Not Changed Test Sensitive Default Entities Data**
There are some Magento default entities are also test sensitive and should not be changed at all time. These include:
Default Admin User, Credentials, Permissions 
Main Website/Main Website Store/Default Store View

Reply:
I'll read through the doc and add some of those snippets 👍

Reply:
@soumyau Added above to fit format of the guide